scenario-compiler/src/main/scala/pl/touk/nussknacker/engine/spel/Typer.scala

```diff
@@ -710,6 +713,15 @@ private[spel] class Typer(
         if tc.runtimeObjType.canBeConvertedTo(Typed[java.util.Collection[_]]) ||
           tc.runtimeObjType.klass.isArray =>
       valid(tc.runtimeObjType.params.headOption.getOrElse(Unknown))
+    case tc: TypedObjectTypingResult if classOf[GenericRecord].isAssignableFrom(tc.runtimeObjType.klass) =>
```

Comment:
Use reflection. Thanks to that we won't be statically binded with a specific avro version in api module